features/admin.impersonation.v1/pages/impersonation-configuration.tsx

```diff
@@ -0,0 +1,257 @@
+    const formRef: MutableRefObject<FormPropsInterface> = useRef<FormPropsInterface>(null);
+
+    const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
+
+    const isReadOnly: boolean = !useRequiredScopes(
+        featureConfig?.server?.scopes?.update,
+    );
+    console.log("readonly" + isReadOnly);
+
+    const dispatch: Dispatch<any> = useDispatch();
+
+    const { t } = useTranslation();
+
+    const [ impersonationConfig, setImpersonationConfig ] =
+        useState<ImpersonationConfigFormValuesInterface>(undefined);
+
+    const {
+        data: originalImpersonationConfig,
+        isLoading: isImpersonationFetchRequestLoading,
+        mutate: mutateImpersonationConfig,
+        error: impersonationConfigFetchRequestError
+    } = useImpersonationConfig();
+
+    useEffect(() => {
+        if (
+            originalImpersonationConfig instanceof IdentityAppsApiException
+            || impersonationConfigFetchRequestError
+        ) {
+            dispatch(
+                addAlert({
+                    description: t("impersonation:notifications." +
+                    "getConfiguration.error.description"),
+                    level: AlertLevels.ERROR,
+                    message: t("impersonation:notifications." +
+                    "getConfiguration.error.message")
+                })
+            );
+            return;
+        }
+
+        if (!originalImpersonationConfig) {
+            return;
+        }
+
+        setImpersonationConfig({
+            enableEmailNotification: originalImpersonationConfig.enableEmailNotification
+        });
+    }, [ originalImpersonationConfig ]);
+
+    /**
+     * Handle Impersonation form submit.
+     */
+    const handleSubmit = (value: boolean) => {
+        const data: ImpersonationConfigPatchInterface[] = [
+            {
+                operation: "REPLACE",
+                path: "/enableEmailNotification",
+                value: value
+            }
+        ];
+
+        updateImpersonationConfigurations(data).then(() => {
+            dispatch(
+                addAlert({
+                    description: t("impersonation:notifications." +
+                    "updateConfiguration.success.description"),
+                    level: AlertLevels.SUCCESS,
+                    message: t("impersonation:notifications." +
+                    "updateConfiguration.success.message")
+                })
+            );
+        }).catch(() => {
+            dispatch(
+                addAlert({
+                    description: t("impersonation:notifications." +
+                    "updateConfiguration.error.description"),
+                    level: AlertLevels.ERROR,
+                    message: t("impersonation:notifications." +
+                    "updateConfiguration.error.message")
+                })
+            );
+        }).finally(() => {
+            mutateImpersonationConfig();
```

Comment:
Let's also add a loading state to show a loading state while the form submission is in progress.

Reply:
- Define a new boolean state variable such as isSubmitting, with its default value set to false.
- When the form is submitting, set its value to true ( `setIsSubmitting(true)`)
- Reset the state to `false`, in the `finally` block ( `setIsSubmitting(false)` )
- Update the `readonly` prop value in the toggle.

```
  readOnly={ isReadOnly || (!isReadOnly && isSubmitting ) }
```